nexus/deployment/src/planner.rs

```diff
@@ -698,16 +718,39 @@ mod test {
             );
         }
 
-        // Arbitrarily choose one of the sleds and mark it non-provisionable.
+        // Arbitrarily choose some of the sleds and mark them non-provisionable
+        // in various ways.
+        let mut sleds_iter = policy.sleds.iter_mut();
+
         let nonprovisionable_sled_id = {
-            let (sled_id, resources) =
-                policy.sleds.iter_mut().next().expect("no sleds");
-            resources.provision_state = SledProvisionState::NonProvisionable;
+            let (sled_id, resources) = sleds_iter.next().expect("no sleds");
+            resources.policy = SledPolicy::InService {
+                provision_policy: SledProvisionPolicy::NonProvisionable,
+            };
+            *sled_id
+        };
+        let expunged_sled_id = {
+            let (sled_id, resources) = sleds_iter.next().expect("no sleds");
+            resources.policy = SledPolicy::Expunged;
+            *sled_id
+        };
+        let decommissioned_sled_id = {
+            let (sled_id, resources) = sleds_iter.next().expect("no sleds");
+            resources.state = SledState::Decommissioned;
             *sled_id
         };
 
-        // Now run the planner with a high number of target Nexus zones.
-        policy.target_nexus_zone_count = 14;
+        // Now run the planner with a high number of target Nexus zones. The
+        // number (16) is chosen such that:
+        //
+        // * we start with 5 sleds
+        // * we need to add 12 Nexus zones
```

Comment:
Tiny nit / question - are we adding 12 or are we still adding 11 (if we started with 5 and are going to 16)?

Reply:
Yeah, it's 11. Thanks for catching it!